src/formatting/formatter/tally.rs

```diff
@@ -0,0 +1,583 @@
+use std::str::FromStr;
+
+use super::*;
+
+#[derive(Debug)]
+enum Style {
+    ChineseCountingRods,
+    ChineseTally,
+    WesternTally,
+    WesternTallyUngrouped,
+}
+
+impl FromStr for Style {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self> {
+        match s {
+            "chinese_counting_rods" => Ok(Style::ChineseCountingRods),
```

Comment:
Maybe some short names should be chosen too.

Reply:
How about abbreviating them to `ccr`, `ct`, `wt`, and `wtu`?

Reply:
Yeah, sure.